### PR TITLE
viewer working

### PR DESCRIPTION
**Pull request by 16bitty**
completed first version of the trace viewer

### CODE REVIEW

File: README.md
Comment (koeninger) [inline README.md]: Maybe worth specifying the file needs to be in the format used by write_traces_binary

File: README.md
Comment (koeninger) [inline README.md:41]: mention clicking on a span will show the tag number and length in cycles

File: viewer/src/Main.rs
Comment (koeninger) [inline viewer/src/Main.rs]: this could be a Span rather than a tuple, right?

File: viewer/src/Main.rs
Comment (koeninger) [inline viewer/src/Main.rs]: this should get fixed before merging since it's a potential correctness problem.
Hopefully it's just one iteration through spans, I think https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.max_by_key looking at .stop should work

File: viewer/src/Main.rs
Comment (koeninger) [inline viewer/src/Main.rs:68]: I'm not sure what SDL does with windows that are larger than someone's screen size, will they still be able to get to the edge of the window to resize it?  Otherwise should probably default this to something safer, like 800.

If you end up introducing some kind of configuration file later, seems like a thing to include there.

File: viewer/src/Main.rs
Comment (koeninger) [inline viewer/src/Main.rs]: I think some of these try_intos may be unnecessary, see what `cargo clippy` reports for this project

File: viewer/src/Main.rs
Comment (koeninger) [inline viewer/src/Main.rs:284]: doesn't need to be this PR, but this might be a place to get some performance back on larger files.  If iterating through the spans takes a significant portion of the time for a "frame" (or longer than it), the sleep could be reduced or skipped.

File: viewer/src/Main.rs
Comment (koeninger) [inline viewer/src/Main.rs]: I try to avoid bare unwrap() in favor of expect where possible.  These ones seem especially likely to trigger if someone misunderstands the format of the command line arguments, they'd probably appreciate a more useful panic message

File: viewer/src/Main.rs
Comment (16bitty) [reply viewer/src/Main.rs:68]: Talked about this behavior in person, will deal with it on a later PR.

File: README.md
Comment (16bitty) [reply README.md]: Thanks! Updated README to mention this.

File: README.md
Comment (16bitty) [reply README.md:41]: Also added this to README.

File: viewer/src/Main.rs
Comment (koeninger) [inline viewer/src/Main.rs:144]: `tag_data: span` should work on its own without unpacking the fields

**Review by koeninger — APPROVED** [review]
lgtm, your call on whether you want to take a look at the frame sleep timing in this PR or a later one.

File: viewer/src/Main.rs
Comment (16bitty) [reply viewer/src/Main.rs:284]: I think I'll leave this for a later PR, since there's a chance that testing a new solution will be a fairly involved process, and it's not necessary at the moment.

**Pull request merged**
